src/prpy/planning/ompl.py

```diff
@@ -163,67 +167,115 @@ def _Plan(self, robot, goal=None, tsrchains=None, timeout=30., shortcut_timeout=
-            epsilon = 1e-6
-
-            # Compute the collision checking resolution as a conservative
-            # fraction of the state space extents. This mimics the logic inside
-            # or_ompl used to set collision checking resolution.
+            # Compute the maximum DOF range, treating circle joints as SO(2).
             dof_limit_lower, dof_limit_upper = robot.GetActiveDOFLimits()
-            dof_ranges = dof_limit_upper - dof_limit_lower
+            dof_ranges = numpy.zeros(robot.GetActiveDOF())
+
+            for index, dof_index in enumerate(robot.GetActiveDOFIndices()):
+                joint = robot.GetJointFromDOFIndex(dof_index)
+                axis_index = dof_index - joint.GetDOFIndex()
+
+                if joint.IsCircular(axis_index):
+                    dof_ranges[index] = 2 * numpy.pi
+                else:
+                    dof_ranges[index] = (dof_limit_upper[index]
+                                       - dof_limit_lower[index])
+
+            # Duplicate the logic in or_ompl to convert the DOF resolutions to
+            # a fraction of the longest extent of the state space.
+            maximum_extent = numpy.max(dof_ranges)
             dof_resolution = robot.GetActiveDOFResolutions()
-            ratios = dof_resolution / dof_ranges
-            conservative_ratio = numpy.min(ratios)
-
-            # Convert the ratio back to a collision checking resolution. Set
-            # RRT-Connect's range to the same value used for collision
-            # detection inside OpenRAVE.
-            longest_extent = numpy.max(dof_ranges)
-            ompl_range = conservative_ratio * longest_extent - epsilon
-
-            ompl_args['range'] = ompl_range
-            logger.debug('Defaulted RRT-Connect range parameter to %.3f.',
-                         ompl_range)
+            conservative_resolution = numpy.min(dof_resolution)
+            conservative_fraction = conservative_resolution / maximum_extent
+
+            if self.multiplier is not None:
+                multiplier = self.multiplier
+            elif self.fraction is not None:
+                multiplier = int(self.fraction / conservative_fraction + 0.5)
+            else:
+                raise ValueError('Either multiplier or fraction must be set.')
+
+            # Then, scale this by the user-supplied multiple. Finally, subtract
+            # a small value to cope with numerical precision issues inside OMPL.
+            ompl_args['range'] = multiplier * conservative_fraction - epsilon
+
         return ompl_args
 
     @ClonedPlanningMethod
     def PlanToConfiguration(self, robot, goal, ompl_args=None, **kw_args):
-        """
-        Plan to a desired configuration with OMPL. This will invoke the OMPL
-        planner specified in the OMPLPlanner constructor.
-        @param robot
-        @param goal desired configuration
-        @return traj
-        """
         ompl_args = self._SetPlannerRange(robot, ompl_args=ompl_args)
```

Comment:
My only other feedback is that `_SetPlannerRange(robot, ompl_args)` is poorly named (it doesn't set the value in the passed `ompl_args` argument, it instead returns a copy with the value set if it doesn't yet exist).  My suggestion would be to rename it to `_CalculatePlannerRange(robot)` which just returns the range float with all dict logic removed, and then replace this line (and the similar line in `PlanToTSR` below) with something like:

```
if ompl_args is None:
    ompl_args = dict()
ompl_args.setdefault('range', _CalculatePlannerRange(robot))
```